src/pages/workspaces/workspace/tools/WorkflowView.js

```diff
@@ -0,0 +1,189 @@
+
+    return div({ style: { display: 'flex' } }, [
+      div({ style: { flex: '1 1 auto', lineHeight: '1.5rem' } }, [
+        div({ style: { color: Style.colors.title, fontSize: 24 } }, name),
+        div(`V. ${methodConfigVersion}`),
+        methodPath && div(`Path: ${methodPath}`),
+        div({ style: { textTransform: 'capitalize' } }, `Data Type: ${rootEntityType}`)
+      ]),
+      div({ style: { flex: '0 0 auto' } }, [
+        buttonPrimary({ disabled: true }, 'Launch analysis')
+      ])
+    ])
+  }
+
+  renderTabs = () => {
+    const { selectedTab } = this.state
+
+    return h(Fragment, [
+      div(
+        { style: { marginTop: '2rem' } },
+        tabs.map(label => {
+          const selected = label === selectedTab
+          const border = `1px solid ${selected ? Style.colors.sectionBorder : Style.colors.section}`
+          return h(Interactive, {
+            as: 'div',
+            style: {
+              display: 'inline-block', position: 'relative', padding: '1rem 1.5rem',
+              fontSize: 16, fontWeight: 500, color: Style.colors.secondary,
+              backgroundColor: selected && Style.colors.sectionHighlight,
+              borderTop: border, borderLeft: border, borderRight: border,
+              borderRadius: '5px 5px 0 0'
+            },
+            onClick: () => this.setState({ selectedTab: label })
+          }, [
+            label,
+            selected && div({
+              style: {
+                // Fractional L/R to make border corners line up when zooming in. Works for up to 175% in Chrome.
+                position: 'absolute', left: 0.4, right: 0.1, bottom: -3, height: 5,
+                backgroundColor: Style.colors.sectionHighlight
+              }
+            })
+          ])
+        })),
+      div(
+        {
+          style: {
+            display: 'flex', padding: '0.3rem',
+            border: `1px solid ${Style.colors.sectionBorder}`,
+            backgroundColor: Style.colors.sectionHighlight,
+            borderBottom: 'unset',
+            borderTopRightRadius: 5,
+            borderTopLeftRadius: selectedTab !== 'Inputs' && 5
+          }
+        },
+        selectedTab === 'WDL' ?
+          // Placeholder to preserve spacing:
+          [div({ style: { fontSize: 12, padding: '0.5rem 0', color: 'transparent', userSelect: 'none' } }, '.')] :
+          tableColumns.map(({ label, width }, idx) => {
+            return div({
+              style: {
+                flex: width ? `0 0 ${width}px` : '1 1 auto',
+                fontWeight: 500, fontSize: 12, padding: '0.5rem 0.8rem',
+                borderLeft: idx !== 0 && Style.standardLine
+              }
+            },
+            label)
+          })
+      )
+    ])
+  }
+
+  renderDetail = () => {
+    const { selectedTab, wdl } = this.state
+
+    if (selectedTab === 'WDL') {
+      if (!wdl) {
+        this.fetchWDL()
```

Comment:
Initiating a fetch in the render path is risky, because there are no guarantees it won't be called multiple times in a row. This would be better handled in lifecycle methods.

Reply:
I didn't want to load the WDL until the user clicks on that tab. What would be a better way to do this?